utils/train_utils.py

```diff
@@ -117,3 +123,69 @@ def test_epoch_ecg(autoencoder, test_loader, device, loss_function, verbose=True
         print(f"====> Test set loss: {avg_loss:.4f}")
 
     return avg_loss
+
+
+def train_epoch_vqvae(
+    autoencoder, train_loader, optimizer, device, log_interval, epoch, loss_function, beta, use_ema, best_train_loss
+):
```

Comment:
```suggestion
def train_epoch_vqvae(
    autoencoder: VQVAE,
    train_loader: DataLoader,
    optimizer: torch.optimizer,
    device: str,
    log_interval: int,
    etc.
):
```

Type hints are always good, but here's one use case I wanted to highlight - I'm reading through your code and having the model class here allows me to jump quickly to its definition (most code editor can do that). 

If there's no type hint with the exact model class, I have to go backward in code through multiple function calls to figure out what exactly is the type of model allowed here.